stl/inc/flat_set

```diff
@@ -180,25 +180,27 @@ public:
     }
 
     _Deriv& operator=(initializer_list<_Kty> _Ilist) {
-        _Clear_guard<_Base_flat_set> _Guard{this};
+        _Clear_guard<container_type> _Guard{_STD addressof(_Mycont)};
         _Mycont.assign(_Ilist);
         _Make_invariants_fulfilled();
         _Guard._Target = nullptr;
         return static_cast<_Deriv&>(*this);
     }
 
     // iterators
+    // for flat_meow, `const_iterator` is required to be convertible to `iterator` (per N4958
+    // [associative.reqmts.general]/6)
     _NODISCARD iterator begin() noexcept {
-        return _Mycont.begin();
+        return cbegin();

```

Comment:
I think we should avoid adding a layer of indirection here to simplify debug codegen and the stepping experience. I don't much care if it's `_Mycont.begin()` or `_Mycont.cbegin()`, but I do think we should call directly rather than via `flat_set::cbegin`. (Four occurrences.)